File: src/main/java/io/vproxy/vfx/manager/internal_i18n/InternalI18n.java
package io.vproxy.vfx.manager.internal_i18n;

import io.vproxy.commons.util.Singleton;

public interface InternalI18n {
    static InternalI18n get() {
        return Singleton.registerIfAbsent(InternalI18n.class, () -> new InternalI18n() {
        });
    }

    default String cancelButton() {
        return "Cancel";
    }

    default String alertInfoTitle() {
        return "Info";
    }

    default String alertWarningTitle() {
        return "Warning";
    }

    default String alertErrorTitle() {
        return "Error";
    }

    default String alertOkButton() {
        return "Ok";
    }

    default String confirmationYesButton() {
        return "Yes";
    }

    default String confirmationNoButton() {
        return "No";
    }

    default String cannotFindAnyDisplay() {
        return "cannot find any display";
    }

    default String stacktraceAlertTitle() {
        return "Exception";
    }

    default String stacktraceAlertHeaderText() {
        return "Exception Thrown";
    }

    default String stacktraceAlertLabel() {
        return "The exception stacktrace was:";
    }

    default String keyChooserLeftMouseButton() {
        return "Left mouse button";
    }

    default String keyChooserRightMouseButton() {
        return "Right mouse button";
    }

    default String keyChooserMiddleMouseButton() {
        return "Middle mouse button";
    }

    default String keyChooserWheelScrollUpButton() {
        return "Wheel scroll up";
    }

    default String keyChooserWheelScrollDownButton() {
        return "Wheel scroll down";
    }

    default String keyChooserDesc() {
        return "Please press the key you want to use, or click the buttons to choose left or right mouse button.";
    }

    default String keyChooserDescWithoutMouse() {
        return "Please press the key you want to use.";
    }

    default String emptyTableLabel() {
        return "No Data";
    }

    default String sceneGroupPreCheckShowSceneFailed() {
        return "An exception occurred when trying to show the scene.";
    }

    default String sceneGroupPreCheckHideSceneFailed() {
        return "An exception occurred when trying to hide the scene.";
    }

    default String loadingCanceled() {
        return "loading process canceled";
    }

    default String globalScreenRegisterFailed() {
        return "Enabling GlobalScreen failed.";
    }
}

File: src-ts/src/manager/internal_i18n/InternalI18n.ts
export class InternalI18n {
  private static instance: InternalI18n = new InternalI18n();

  static get(): InternalI18n { return InternalI18n.instance; }
  static set(i18n: InternalI18n): void { InternalI18n.instance = i18n; }

  cancelButton(): string { return 'Cancel'; }
  alertInfoTitle(): string { return 'Info'; }
  alertWarningTitle(): string { return 'Warning'; }
  alertErrorTitle(): string { return 'Error'; }
  alertOkButton(): string { return 'Ok'; }
  confirmationYesButton(): string { return 'Yes'; }
  confirmationNoButton(): string { return 'No'; }
  cannotFindAnyDisplay(): string { return 'cannot find any display'; }
  stacktraceAlertTitle(): string { return 'Exception'; }
  stacktraceAlertHeaderText(): string { return 'Exception Thrown'; }
  stacktraceAlertLabel(): string { return 'The exception stacktrace was:'; }
  keyChooserLeftMouseButton(): string { return 'Left mouse button'; }
  keyChooserRightMouseButton(): string { return 'Right mouse button'; }
  keyChooserMiddleMouseButton(): string { return 'Middle mouse button'; }
  keyChooserWheelScrollUpButton(): string { return 'Wheel scroll up'; }
  keyChooserWheelScrollDownButton(): string { return 'Wheel scroll down'; }
  keyChooserDesc(): string {
    return 'Please press the key you want to use, or click the buttons to choose left or right mouse button.';
  }
  keyChooserDescWithoutMouse(): string {
    return 'Please press the key you want to use.';
  }
  emptyTableLabel(): string { return 'No Data'; }
  sceneGroupPreCheckShowSceneFailed(): string {
    return 'An exception occurred when trying to show the scene.';
  }
  sceneGroupPreCheckHideSceneFailed(): string {
    return 'An exception occurred when trying to hide the scene.';
  }
  loadingCanceled(): string { return 'loading process canceled'; }
  globalScreenRegisterFailed(): string { return 'Enabling GlobalScreen failed.'; }
}
